app/services/schools/training_programme.rb

```diff
@@ -1,6 +1,47 @@
 module Schools
   class TrainingProgramme
-    def initialize(school)
+    def initialize(school:, registration_period_id:)
+      @school = school
+      @registration_period_id = registration_period_id.to_i
+    end
+
+    def training_programme
+      return not_yet_known unless partnership_exists?
+      return provider_led if mentors_at_school?
```

Comment:
I'm not sure we are we meeting this point here:
```
1+ school partnerships and 1+ ECT/Mentors but no provider_led ECTs: school_led
```
as we aren't checking the ects?